pkg/odo/cli/component/deploy.go

```diff
@@ -64,7 +67,29 @@ func (do *DeployOptions) Complete(name string, cmd *cobra.Command, args []string
 
 // Validate validates the push parameters
 func (do *DeployOptions) Validate() (err error) {
-	// TODO: Validate the value of tag and any user parameteres.
+
+	// Validate the --tag
+	if do.tag == "" {
+		return errors.New("odo deploy requires a tag, in the format <registry>/namespace>/<image>")
+	}
+	var splitTag = strings.Split(do.tag, "/")
+	if len(splitTag) != 3 {
+		return errors.New("invalid tag: odo deploy reguires a tag in the format <registry>/namespace>/<image>")
+	}
+
+	// Valid characters for the registry, namespace, and image name
+	characterMatch := regexp.MustCompile(`[a-zA-Z0-9\.\-:_]*`)
+	for _, element := range splitTag {
+		elementMatch := characterMatch.MatchString(element)
+		if !elementMatch {
+			return errors.New("invalid tag: " + element + " in the tag contains an illegal character. It must only contain alphanumerical values, periods, colons, underscores, and dashes.")
+		}
+		// The registry, namespace, and image, cannot end in '.', '-', '_',or ':'
+		if strings.HasSuffix(element, ".") || strings.HasSuffix(element, "-") || strings.HasSuffix(element, ":") || strings.HasSuffix(element, "_") {
+			return errors.New("invalid tag: " + element + " in the tag has an invalid final character. It must end in an alphanumeric value.")
+		}
+	}
```

Comment:
I would put this into a function, and create unit tests for it. Can be done in a different PR :)